Tests/SymbolKitTests/SymbolGraph/AnyScalarTests.swift

```diff
@@ -0,0 +1,89 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import XCTest
+import SymbolKit
+
+class AnyScalarTests: XCTestCase {
+    func testAnyScalarCanBeDecoded() throws {
+        let decoder = JSONDecoder()
+        var scalar = try decoder.decode(SymbolGraph.AnyScalar.self, from: "3".data(using: .utf8)!)
+        if case let SymbolGraph.AnyScalar.integer(value) = scalar {
+            XCTAssertEqual(value, 3)
+        } else {
+            XCTFail("Unable to extract integer AnyScalar from JSON string")
+        }
+        
+        scalar = try decoder.decode(SymbolGraph.AnyScalar.self, from: "3.5".data(using: .utf8)!)
+        if case let SymbolGraph.AnyScalar.float(value) = scalar {
+            XCTAssertEqual(value, 3.5)
+        } else {
+            XCTFail("Unable to extract float AnyScalar from JSON string")
+        }
```

Comment:
I assume this will work out, but i wonder about type collisions between floats and integers. Will `3.0` correctly decode as a float?

Reply:
You're right, it will decode as an integer. I don't think there would be a client impact in practice, but I'll keep of this.